src/Samples/Basic/Temporal.Samples.DesignProcess/Part1_2_PermanentWorkflow.cs

```diff
@@ -0,0 +1,132 @@
+            public void SetAddressee(SpeechRequest input)
+            {
+                _setAddresseeName.TrySetResult(input?.Text);                
+            }
+
+            [WorkflowSignalHandler]
+            public void Cancel()
+            {
+                while (!_setAddresseeName.TrySetCanceled())
+                {
+                    _setAddresseeName = new TaskCompletionSource<string>();
+                }                
+            }
+        }
+
+        /// <summary>A sample activity implementation.</summary>
+        public static class Speak
+        {
+            public static Task GreetingAsync(SpeechRequest input, WorkflowActivityContext activityCtx)
+            {
+                Console.WriteLine($"[{activityCtx.ActivityTypeName}] {input.Text}");
+                return Task.CompletedTask;
+            }
+        }
+
+        public static class RemoteApiNames
+        {
+            public static class Activities
+            {
+                public const string SpeakGreeting = "SpeakAGreeting";
+            }            
+        }
+
+        /// <summary>
+        /// Parameters to workflow APIs (main method, signal & query parameters) and to activities must implement <see cref="IDataValue" />.
+        /// In some specialized cases where it is not possible, the raw (non-deserialized) payload may be accessed
+        /// directly (e.g. <see cref="Part4_1_BasicWorkflowUsage" /> and <see cref="Part4_2_BasicWorkflowUsage_MultipleWorkers" />).
+        /// </summary>
+        public class SpeechRequest : IDataValue
+        {
+            public string Text { get; set; }
+
+            public SpeechRequest(string text)
+            {
+                Text = text;
+            }
+        }
+
+        public static void Main(string[] args)
+        {
+            // Use the ASP.Net Core host initialization pattern:
+            // 'UseTemporalWorkerHost' will configure all the temporal defaults
+            // and also apply all the file-based application config files.
+            // Configuraton will automatically be persisted via side affects as appropriate before being passed to the workflow implmentation.
+            // (Config files are treated elsewhere.)            
+
+            IHost appHost = Host.CreateDefaultBuilder(args)
+                    .UseTemporalWorkerHost()
+                    .ConfigureServices(serviceCollection =>
+                    {
+                        serviceCollection.AddTemporalWorker()
+                                .Configure(temporalWorkerConfig =>
+                                {
+                                    temporalWorkerConfig.TaskQueueMoniker = "Some Queue";
+                                });
+
+                        serviceCollection.AddWorkflowWithAttributes<SayGreetingWorkflow>();
+
+                        serviceCollection.AddActivity<SpeechRequest>(RemoteApiNames.Activities.SpeakGreeting, Speak.GreetingAsync);
```

Comment:
Is it just for demonstration purposes that the activity name is different than the method name? From an implementer POV (and therefore from a best-practices/example POV), I'd probably name them the same.